### ISSUE TITLE
M1-12 Proactive onboarding of services and models

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#694]
Original Wrike title: "Increase User Base & Available Resources"

Proactive onboarding of services and models, with focus on SPARC and ANS neuroscience needs [(e.g., by contacting authors of SPARC-relevant modeling publications)](https://sparc.science/data?type=simulation&selectedFacetIds=0&skip=0&limit=50).
Deliverable: Services and models on o²S²PARC. Targeted number per quarter: 1-2 services per quarter and 2-3 models per quarter (assuming models can be onboarded using existing services capabilities, e.g. Jupyterlabs).
Deadline: Month 1–12

(NIH comment to SIM-Core: “Bring more autonomic neuroscience models and simulations into oSPARC. It was terrific to see oSPARC being used for high-impact modeling studies this year (somatosensory and brain stimulation), but those studies were not within SPARC’s domain area”; SC feedback to SIM-Core: “Build and feature models generated from actual SPARC data.”)
Relates as well to Publication on Portal (﻿#684)

See ﻿[Y6 SIM – 7. Sustainability: Focus on Sustainability](https://www.wrike.com/open.htm?id=916192754)﻿ for related milestones.

[Wrike M5.2](https://www.wrike.com/workspace.htm?acc=3203588#/task-view?id=916307072&pid=916192224&cid=720993193)

```[tasklist]
### Tasks
- [ ] New version sim_gui-app/web (deprecated)
- [ ] New version tvb-app/web (deprecated)
- [ ] EIT/Kirill's use-case: Update EIDORS Service @elisabettai
- [ ] EIT/Kirill's use-case: Additional mesher service? @elisabettai
- [ ] https://github.com/ITISFoundation/osparc-issues/issues/942
```

```[tasklist]
### Jelly Beans
- [x] CC lab models: finish service creation and submit for publication on the Portal
- [x] Codeathon 2022: submit 6 studies for publication on the Portal
- [x] Publish BIOS (ask authors before)
- [x] Follow up with OA, they have studies on oSPARC that could be published
- [x] Follow up with ZD: he said their work might be ready for Jan/Feb
- [x] Follow up with Raj (AN): AN was interested in putting models on oSPARC
- [x] Maybe, depending on outcome of above: update ASCENT with their requests
```

```[tasklist]
### Pastel de Nata
- [x] Check progress with Omkar, support for service creation (service creates an opencor config) if needed
- [x] Follow up with Zach: he said their work might be ready for Jan/Feb
- [x] Maybe, depending on outcome of above: update ASCENT with their requests
- [x] Other: call on Fri 19 May with Kirill
- [x] Call with Sergey
```

```[tasklist]
### Sprint [WaterMelon](https://github.com/ITISFoundation/osparc-simcore/milestone/46)
- [x] New version of jupyter-chaste @elisabettai https://github.com/ITISFoundation/jupyter-chaste/pull/1
- [x] Turn [opencor_Config_osparc_service](https://github.com/OmkarAthavale/opencorConfig_osparc_service) into a Service @elisabettai
- [x] Onboard k-Wave (simulator for acoustic simulations in C++, GPU-optimized) @elisabettai https://github.com/ITISFoundation/k-wave/pull/2
- [x] Cardiovascular Simulink/MATLAB model from RP (SPARC group) @elisabettai https://github.com/ITISFoundation/CardiovascularControl-DBI/pull/1
```

```[tasklist]
### Sprint Sundae
- [ ] List above a task for every new service to be upgraded @mrnicegyu11
- [x] New version of kember-viewer (deprecated) CRITICAL @odeimaiz. Kember viewer has been removed, there's no replacement yet
- [ ] Add deprecation mark to all legacy services (based on excel list) @mrnicegyu11
- [x] Make "generic" plotting service: use-case: visualize output of [CardioControl-DBI service](https://github.com/ITISFoundation/CardiovascularControl-DBI). It should be accessible to Guest users as well. @elisabettai https://github.com/ITISFoundation/voila-viewer
- [ ] https://github.com/ITISFoundation/osparc-issues/issues/998
- [ ] Sergey LU: Make 3 services out  of Sergey's Matlab scripts @elisabettai
```

**Comment by elisabettai** [comment]
## Update on sprint [Vaporwave](https://github.com/ITISFoundation/osparc-issues/blob/master/reviews/20221007_Vaporwave.md)

### Done
- Created 5 new studies with tutorials from SPARC Fair Codeathon

### Ongoing
- 6th study from SPARC Fair Codeathon, blocked by problem with JupyterLab widget (ipyleaflet), opened an [issue in ipyleaflet repository](https://github.com/jupyter-widgets/ipyleaflet/issues/1062). 

### Open
- Publish studies on the Portal: I will attend K-Core office hours on Oct 13

**Comment by elisabettai** [comment]
## Goal for sprint Mithril 
### Onboard new services
- JupyeterLab with tvb-recon and other neuron packages (e.g. FSL), as requested by TN and EN.

### Onboard new models (as comp. services)
- New (updated) cardiac model from CC lab. We received source code in C++, we need to check what is different from existing models and publish new versions/models as appropriate.

**Comment by elisabettai** [comment]
The NIH would like to have more (oSPARC) computational studies published [in this section](https://sparc.science/data?type=simulation) of the SPARC portal. 

EI and EN met to define priorities, for this sprint will try to:

## Goal for sprint [Jelly Beans](https://github.com/ITISFoundation/osparc-issues/milestone/41)
- [ ] CC lab models: finish service creation and submit for publication on the Portal 
- [ ] Codeathon 2022: submit 6 studies for publication on the Portal
- [ ] Follow up with OA, they have studies on oSPARC that could be published
- [ ] Follow up with ZD: he said their work might be ready for Jan/Feb
- [ ] Follow up with Raj (AN): AN was interested in putting models on oSPARC
- [ ] Maybe, depending on outcome of above: update ASCENT with their requests

**Comment by elisabettai** [comment]
## Update for sprint [Watermelon](https://github.com/ITISFoundation/osparc-issues/blob/master/reviews/20230707_Watermelon.md)

### Done
- Update legacy interactive services: new jupyter-chaste service and open-sourced the [code](https://github.com/ITISFoundation/jupyter-chaste).
- New service: [k-Wave](https://github.com/ITISFoundation/k-wave) (simulator for acoustic simulations in C++, GPU-optimized). Awaiting feedback from users
- New service: [Cardiovascular Simulink/MATLAB model](https://github.com/ITISFoundation/CardiovascularControl-DBI) from RP (SPARC group) + [template](https://osparc.io/study/65a52908-1b13-11ee-9cd1-02420a0b1bed) to be published on the SPARC Portal.

### To do
See tasks list above

**Comment by elisabettai** [comment]
## Update for sprint [Sundae](https://github.com/ITISFoundation/osparc-issues/blob/master/reviews/20230814_Sundae.md)

### Done
- Made a reusable voila viewer service, so we don't have to craft every time a new service for plotting simple graphs and share templates with anonymous users. See code and instructions [here](https://github.com/ITISFoundation/voila-viewer).
- Made a template on master to assess security of the viewer for anonymous users, see [this case](https://github.com/ITISFoundation/voila-viewer/issues/2).
- While we assess the security of the voila-viewer, a [custom viewer](https://github.com/ITISFoundation/cardiocontro-dbi-viewer) was made for the Simulink Cardio model we recently onboarded, so that we can share it with anonymous users without concern. The provider of the model approved it for publication on the SPARC Portal. 
- Updated the jupyter-ascent service with the last small/cosmetic/usability improvements received from NP.
- Received approval from NP to publish the three ASCENT templates on the Portal. Metadata for submission to the Portal were prepared and shared with NP. The three templates are already public on oSPARC.

### In progress
- Test the oSPARC ASCENT implementation with a dataset from the portal. Blocked for the moment for a weird COMSOL error that we're debugging.

### Todo:
- Run MATLAB code from prof. Makaroff and turn it into comp. services on oSPARC.
- Check what we can do already for Kirill in terms of onboarding.

**Comment by elisabettai** [comment]
## Update for sprint [Baklava](https://github.com/ITISFoundation/osparc-issues/blob/master/reviews/20230908_Baklava.md)

### Done
- Cardiovascular System model from MH (NeuHeart): a [template](https://osparc.io/#/study/fa885632-4b04-11ee-869a-02420a0bd26e) is now public on osparc.io (needs login because it's a JupyterLab. It has also been submitted for Publication on the SPARC Portal.

### In progress
- FMM-LU Services (Matlab) code from prof. Makaroff. 
- - Code runs locally in Matlab, apart from LU decomposition that requires at 128GB of RAM.
- - Done first draft of modularization

### Blocked
- Test the oSPARC ASCENT implementation with a dataset from the portal. https://github.com/ITISFoundation/osparc-issues/issues/998. Weird COMSOL error and ASCENT devs didn't help much so far.

**Comment by elisabettai** [comment]
Closing this one and moved the remaining tasks to the new issue for Y7 #1093 
The list of what has been onboarded in Y6 is in the wrike ticket linked in the description.
